source/level/TestGenState.hx

```diff
@@ -11,20 +18,54 @@ class TestGenState extends FlxState {
 	private static var LEVEL_HEIGHT(default, never):Int = 32;
 	private static var LEVEL_WIDTH(default, never):Int = 32;
 
+	private var map:FlxTilemap;
+
+	private var targetTank:Tank;
+	private var enemyTank:Tank;
+
 	override public function create() {
 		super.create();
 
 		var gen = new LevelGenerator(20, 20);
 		var bitmap = gen.generateLevel();
 
-		var map = new FlxTilemap();
+		map = new FlxTilemap();
 		trace(bitmap);
 		map.loadMapFrom2DArray(bitmap, AssetPaths.Wall__png, TILE_WIDTH, TILE_HEIGHT);
 
 		add(map);
+
+		setUpHorizontalTank();
+		targetTank = setUpPlayerTank();
+		enemyTank = setUpDynamicTank(targetTank, map);
+	}
+
+	private function setUpHorizontalTank() {
+		var tank = new Tank(500, 50);
+		tank.setControllers(new HorizontalMoveController(tank), new SpinShootController(tank));
+		add(tank.getAllSprites());
+		return tank;
+	}
+
+	private function setUpPlayerTank() {
+		var playerTank = TankFactory.NewPlayerTank(100, 50);
+		add(playerTank.getAllSprites());
+		return playerTank;
+	}
+
+	private function setUpDynamicTank(target, tileMap) {
+		var dynamicTank = new Tank(50, 500);
+		dynamicTank.setControllers(new DynamicMovementController(dynamicTank, target, tileMap),
+			new AutoAimController(dynamicTank, target));
```

Comment:
This works fine, but I wanted to mention that this is the bit that might go in the TankFactory class, along with code to set up any special graphic stuff for this particular variety of tank.

It would all function the same, it just gives us one central location (inside `TankFactory` that defines what a "Dynamic Tank" really is.

Reply:
Sounds good thanks for the reminder--I moved it over to TankFactory which made everything look more organized. So now in the TestGenState, the function for making a new dynamic tank has additional parameters besides position, it also needs the player tank object and map object unlike the other types of tanks.